osidb/migrations/0086_migrate_flaw_flag_nist_cvss_validation.py

```diff
@@ -0,0 +1,101 @@
+"""
+Written manually on 2023-07-17
+
+Copy FlawMeta NIST_CVSS_VALIDATION into Flaw field nist_cvss_validation.
+
+To prevent out of memory issues, FlawMeta are iterated using .iterator()
+so that FlawMeta instances are not cached. To prevent further out of
+memory issues and runtime efficiency, the migration uses a generator,
+islice and bulk_create as recommended by
+https://docs.djangoproject.com/en/3.2/ref/models/querysets/#bulk-create
+
+This migration should happen in lockstep with adding bzimport for Flaw
+field nist_cvss_validation. This ensures that any future operations performed
+using the new field do not result in data loss. (For a hypothetical example,
+if the flag value was recorded only in FlawMeta but not in the Flaw field,
+bbsync based on the field would effectively delete the flag from bugzilla,
+and that would also delete the FlawMeta-based flag on next bzimport.)
+"""
+
+from django.db import migrations
+from itertools import islice
+
+
+def generate_flag_values(apps):
+    """
+    Generates pairs of (Flaw instance, value of flag nist_cvss_validation for that Flaw).
+    """
+    FlawMeta = apps.get_model("osidb", "FlawMeta")
+    for flawmeta in FlawMeta.objects.filter(type="NIST_CVSS_VALIDATION").iterator():
+        flaw = flawmeta.flaw
+        flag_value = flawmeta.meta_attr["status"]
+        yield flaw, flag_value
+
+
+def bulk_update(apps, flaws_flags):
+    """
+    Updates flag nist_cvss_validation on Flaws provided in the argument.
+    """
+    Flaw = apps.get_model("osidb", "Flaw")
+    flaws = []
+    for flaw, flag in flaws_flags:
+        flaw.nist_cvss_validation = flag
+        flaws.append(flaw)
+    Flaw.objects.bulk_update(flaws, ["nist_cvss_validation"])
+
+
+def run_batches(apps, batch_size, flag_val_generator):
+    """
+    Slices all Flaws that should be updated into batches and executes
+    the bulk Flaw model updates.
+    """
+    while True:
+        # Limiting memory usage by using a generator and slicing recommended in
+        # https://docs.djangoproject.com/en/3.2/ref/models/querysets/#bulk-create
+        batch_flaws_flags = list(islice(flag_val_generator, batch_size))
+        if not batch_flaws_flags:
+            break
+        bulk_update(apps, batch_flaws_flags)
+
+
+def forwards_func(apps, schema_editor):
+    """
+    For all Flaws that have the bugzilla flag nist_cvss_validation set, this
+    migration sets the value of the field nist_cvss_validation to be identical
+    to existing FlawMeta NIST_CVSS_VALIDATION.
+    """
+    batch_size = 1000
+
+    run_batches(apps, batch_size, generate_flag_values(apps))
+
+
+def backwards_func(apps, schema_editor):
```

Comment:
Because it's documented in https://docs.djangoproject.com/en/3.2/ref/migration-operations/. If we never want reversibility, please discuss with project owner, but seeing as he's already approved, I'm not going to engage in this proactively.

For `FlawAcknowledgment` see https://github.com/RedHatProductSecurity/osidb/blob/master/osidb/migrations/0082_migrate_flawacknowledgment_flawmeta.py#L69-L76.

Reply:
Discussed. No action.